### ISSUE TITLE
TFRecord format requirement for multiple instances of the same class on one training image

### ISSUE THREAD

**Issue by yanfengliu** [tensorflow/models#1948]
I am trying to train a Faster R-CNN on player detection using the new Object Detection API, but I do not quite understand the process of creating a TFRecord file for that. I am aware of the Oxford and VOC dataset examples and the scripts to create TFRecord files, and they work fine if **there is only one object in a training image**, which is what I see in all of the official example. However, I am trying to annotate 20+ objects of the class 'player' in an image. Should they be put in different `XML` files before they are converted to `TFRecord` or just one `XML` per image? Thanks!

**Comment by michaelisard** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by yanfengliu** [comment]
In case anyone else here is looking for an answer on this matter, I think this [page](https://github.com/tensorflow/models/blob/master/object_detection/g3doc/using_your_own_dataset.md) explained it in the code comment that `xmin`, `xmax`, `ymin`, `ymax`, `classes_text`, and `classes` should all be lists with one value per bounding box. Therefore, for each image only one XML file is needed in order to create a TFRecord.

**Comment by rajarsheem** [comment]
The above link is broken. [Here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/using_your_own_dataset.md) is the updated one. But one question I have though is how ```create_cat_tf_example``` in the example change if the image has multiple objects with corresponding bboxes.

**Comment by rajarsheem** [comment]
The above link is broken. [Here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/using_your_own_dataset.md) is the updated one.

**Comment by innjoshka** [comment]
I'm the new one here but I have the same question that was asked in this issue. I created [stackoverflow question] (https://stackoverflow.com/questions/49065432/tfrecord-format-for-multiple-instances-of-the-same-or-different-classes-on-one-t) but I didn't got any answer by the way I followed all the way that @yanfengliu said. Are these lists for multiple object values work fine/not or am I missing something? Please help to find the answer.  

The problem is that there is no any examples available where I can see the format that works with multiple instances of the same class on one training image

**Comment by nathanaherne** [comment]
@innjoshka I had the same questions you had when I wanted to train on multiple objects of the same class in the same image. In fact my image has multiple objects of the same class, multiple times per image. I didn't get an answer to my question about this on StackOverflow or Reddit. How I solved my question was by inferring from others training that it was possible. In instances of training to detect certain supermarket products or satellite photos, there was no other way to train than to label all the objects in an image and then feed that to your model of choice. I tried that (I use Tensorflow) and I am getting reliable detections.
